### ISSUE TITLE
Cannot use python ** operator on ColumnClause

### ISSUE THREAD

**Issue by Rolfdv** [sqlalchemy/sqlalchemy#8579]
### Describe the use case

Currently, most python operations are specified for `sqlalchemy.sql.column` objects, e.g. multiplication:

```python
from sqlalchemy.sql import column

column("someColumn") * 2
$ <sqlalchemy.sql.elements.BinaryExpression object at 0x7f873a24ce20> 
```

However, this functionality appears to be missing for raising to the power:
```python
column("someColumn") ** 2

$ TypeError: unsupported operand type(s) for ** or pow(): 'ColumnClause' and 'int'
```

Similarly, raising to the power of another column also results in a TypeError. Defining __pow__ and __rpow__ in `sqlalchemy.sql.operators.py` would resolve this issue

### Databases / Backends / Drivers targeted

-

### Example Use

```python
column("a") ** column("bb")
```
`->`
```sql
pow(a, b)
```

**Comment by CaselIT** [comment]
I think pow is an alias of power. At least on pg pow is not documented, but power is. https://www.postgresql.org/docs/current/functions-math.html#FUNCTIONS-MATH-FUNC-TABLE
I believe mysql is the same.  So I think we could just map to power instead of pow. 

For sqlite it seems it depend on how it's compiled.
I get this:
```
SQLite version 3.39.2 2022-07-21 15:24:47
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> select pow(2, 16);
Parse error: no such function: pow
  select pow(2, 16);
         ^--- error here
sqlite> select power(2, 16);
Parse error: no such function: power
  select power(2, 16);
         ^--- error here
sqlite>
```

**Comment by zzzeek** [comment]
hmmm that's bad.   this feels a little niche to be one of the main operators

**Comment by CaselIT** [comment]
from https://www.sqlite.org/lang_mathfunc.html

> The math functions shown below are part of the [SQLite amalgamation source file](https://www.sqlite.org/amalgamation.html) but are only active if the amalgamation is compiled using the [-DSQLITE_ENABLE_MATH_FUNCTIONS](https://www.sqlite.org/compile.html#enable_math_functions) compile-time option.

**Comment by zzzeek** [comment]
my local sqlite here has it so I would assume this compile feature is common in packaged sqlite builds.

**Comment by CaselIT** [comment]
The one I've tested above is from conda on windows.

I have not checked to see if debian and the docker python image have it

**Comment by CaselIT** [comment]
Note that I don't think that the fact that pow is missing from some installations of sqlite is a reason not to have this

**Comment by zzzeek** [comment]
yeah I just feel like adding more operator overloads is not a good idea at this point.   support for pow() with `**` can be implemented now with custom types but it feels a little too "miscellaneous" to be hardwired into `**`, although, what else would `**` do is a good argument in favor.

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **main** branch:

**Add pow operator support** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5836
